### ISSUE TITLE
Override a protected @Composable method leads to Compilation Failed on iOS target

### ISSUE THREAD

**Issue by ssttkkl** [JetBrains/compose-multiplatform#4055]
**Describe the bug**
Override a protected `@Composable` method leads to Compilation failed on iOS target. 

**Affected platforms**
- iOS

**Versions**
- Kotlin version*: 1.9.21 
- Compose Multiplatform version*: 1.6.0-alpha01

**To Reproduce**

Just add this piece of code and run gradle task `linkPodDebugFrameworkIosSimulatorArm64`. 

```kotlin
package io.ssttkkl.demo

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable

abstract class AbstractGreeter {
    @Composable
    protected abstract fun Greeting()
}

class Greeter(val target: String) : AbstractGreeter() {
    @Composable
    override fun Greeting() {
        Text("Hello, $target! ")
    }
}
```

Got this error message:

```
e: Compilation failed: no implementation for FUN MISSING_DECLARATION name:Greeting visibility:public modality:FINAL <> () returnType:kotlin.Nothing
in io.ssttkkl.demo.AbstractGreeter

 * Source files: 
 * Compiler version: 1.9.21
 * Output kind: FRAMEWORK
```

**Expected behavior**
This code should be built successfully.

**Comment by shishkin-pavel** [comment]
should be fixed since 1.5.8.1

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4055">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
